src/index.js

```diff
@@ -131,8 +131,8 @@ class Service {
 
     let { query, options } = multiOptions(id, params, this.id);
 
-    // We can not update the id
-    delete data[this.id];
```

Comment:
But remember, a hook with `delete hook.data.id` will create orphan resources, so it really needs to be `hook.data.id = id`...